### PR TITLE
feat(commerce): use new product and pagination properties

### PR DESCRIPTION
**Pull request by Spuffynism**
This PR follows the changes done in [CAPI-696](https://coveord.atlassian.net/browse/CAPI-696). ~Will also create a PR in barca.~ Created here: https://github.com/coveo/barca-sports/pull/219

~Keeping it as draft until [CAPI-696](https://coveord.atlassian.net/browse/CAPI-696) reaches production.~ [CAPI-696](https://coveord.atlassian.net/browse/CAPI-696) has now reached production.

[CAPI-752]

[CAPI-696]: https://coveord.atlassian.net/browse/CAPI-696?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CAPI-696]: https://coveord.atlassian.net/browse/CAPI-696?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CAPI-752]: https://coveord.atlassian.net/browse/CAPI-752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/api/commerce/common/product.ts [hunk 1]
```diff
@@ -81,12 +86,12 @@ export interface Product {
   additionalFields: Record<string, unknown>;
 
   /**
-   * The child results of the product, fetched through [product grouping](https://docs.coveo.com/en/l78i2152).
+   * The child products of the product, fetched through [product grouping](https://docs.coveo.com/en/l78i2152).
    */
-  childResults: Product[];
+  childProducts: ChildProduct[];
```
Comment (Spuffynism) [inline packages/headless/src/api/commerce/common/product.ts:91]: @agillesCoveo @olamothe what would you think of naming this `children` instead, and naming `totalNumberOfChildrenProducts` to `totalNumberOfChildren` (on the API and in headless)? `childProducts` feels redundant to me.

Comment (agillesCoveo) [reply packages/headless/src/api/commerce/common/product.ts:91]: Make sense to me! Fyi I'm creating a ChildProduct model as well (to remove the children list + total number of children in the children themselft)

Comment (olamothe) [reply packages/headless/src/api/commerce/common/product.ts:91]: Yes all good

Comment (Spuffynism) [reply packages/headless/src/api/commerce/common/product.ts]: Applied this change in https://github.com/coveo/ui-kit/pull/3785/commits/17d9d9634f7c0f06c1bf48730a7ef0c834fc0454

**Review by agillesCoveo — APPROVED** [review]
Looks great! Just not sure about the removing of recommendation given the context of this PR mais I guess you harmonize product across the different functionalities?

**Comment by Spuffynism** [comment]
@agillesCoveo To give a little more context, the change of properties made me realize that we had forgotten to use the proper mock builders for recs. This is why this PR contains changes around recs.

**Review by olamothe — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/features/commerce/product-templates/product-templates-manager.ts [hunk 2]
```diff
@@ -1,31 +1,26 @@
-import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {Product} from '../../../api/commerce/common/product';
```
Comment (Spuffynism) [inline packages/headless/src/features/commerce/product-templates/product-templates-manager.ts:1]: @fbeaudoincoveo @olamothe I'm not too use what this is used for but I had to change the type from `ProductRecommendation` to the new `Product`

**Pull request merged**